### PR TITLE
Strict read only mode

### PR DESCRIPTION
**Pull request by hg-ms**
A strict read only storage mode that can be enabled/disabled and does not allow any write actions

### CODE REVIEW

File: storage/storage/src/main/java/one/microstream/storage/types/StorageWriteControllerReadOnlyMode.java [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+			this.writeController == this.readOnlyController
+		);
+	}
+	
+	
+	///////////////////////////////////////////////////////////////////////////
+	// methods //
+	////////////
+	
+	public void enableReadOnly(final boolean enableReadOnly)
```
Comment (fh-ms) [inline storage/storage/src/main/java/one/microstream/storage/types/StorageWriteControllerReadOnlyMode.java]: Probably better to use getter/setter style naming, here setReadOnly(boolean)

File: storage/storage/src/main/java/one/microstream/storage/types/StorageWriteControllerReadOnlyMode.java [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+		}
+		else
+		{
+			this.writeController = this.defaultController;
+		}
+		
+		logger.info("Read-only mode enabled: {}", this.writeController == this.readOnlyController);
+	}
+	
+	public boolean readOnlyEnabled()
```
Comment (fh-ms) [inline storage/storage/src/main/java/one/microstream/storage/types/StorageWriteControllerReadOnlyMode.java]: Probably better to use getter/setter style naming, here isReadOnly()

File: storage/storage/src/main/java/one/microstream/storage/types/StorageWriteControllerReadOnlyMode.java [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+	 * that is used if the read only mode is disabled.
+	 */
+	public StorageWriteControllerReadOnlyMode(final WriteController writeController)
+	{
+		super();
+		this.defaultController = writeController;
+		this.readOnlyController = WriteController.Disabled();
+		this.writeController = this.readOnlyController;
+		
+		logger.info("Created SwitchableStorageWriteController with Storage read-only mode enabled: {}!",
```
Comment (fh-ms) [inline storage/storage/src/main/java/one/microstream/storage/types/StorageWriteControllerReadOnlyMode.java]: StorageWriteControllerReadOnlyMode != SwitchableStorageWriteController

File: storage/storage/src/main/java/one/microstream/storage/types/StorageWriteControllerReadOnlyMode.java (hunk 1 above)
Comment (hg-ms) [reply storage/storage/src/main/java/one/microstream/storage/types/StorageWriteControllerReadOnlyMode.java]: ok, done

File: storage/storage/src/main/java/one/microstream/storage/types/StorageWriteControllerReadOnlyMode.java (hunk 2 above)
Comment (hg-ms) [reply storage/storage/src/main/java/one/microstream/storage/types/StorageWriteControllerReadOnlyMode.java]: ok, done

File: storage/storage/src/main/java/one/microstream/storage/types/StorageWriteControllerReadOnlyMode.java (hunk 3 above)
Comment (hg-ms) [reply storage/storage/src/main/java/one/microstream/storage/types/StorageWriteControllerReadOnlyMode.java]: ok, done

**Review by fh-ms — APPROVED** [review]

**Pull request merged**